Gems/ROS2/Code/Include/ROS2/Frame/ROS2FrameEditorComponent.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzFramework/Components/TransformComponent.h>
+#include <AzToolsFramework/ToolsComponents/EditorComponentBase.h>
+#include <ROS2/Frame/NamespaceConfiguration.h>
+#include <ROS2/Frame/ROS2FrameComponent.h>
+#include <ROS2/Frame/ROS2FrameConfiguration.h>
+#include <ROS2/Frame/ROS2Transform.h>
+#include <ROS2/ROS2GemUtilities.h>
+
+namespace ROS2
+{
+    //! This component marks an interesting reference frame for ROS2 ecosystem.
+    //! It serves as sensor data frame of reference and is responsible, through ROS2Transform, for publishing
+    //! ros2 static and dynamic transforms (/tf_static, /tf). It also facilitates namespace handling.
+    //! An entity can only have a single ROS2Frame on each level. Many ROS2 Components require this component.
+    //! @note A robot should have this component on every level of entity hierarchy (for each joint, fixed or dynamic)
+    class ROS2FrameEditorComponent
+        : public AzToolsFramework::Components::EditorComponentBase
+        , public ROS2FrameComponentBus::Handler
+        , public AZ::EntityBus::Handler
+    {
+    public:
+        AZ_COMPONENT(ROS2FrameEditorComponent, "{f76d6f29-73c3-40b2-bcc2-47fc824c25df}", AzToolsFramework::Components::EditorComponentBase);
+
+        ROS2FrameEditorComponent() = default;
+        ~ROS2FrameEditorComponent() = default;
+        //! Initialize to a specific frame id
+        ROS2FrameEditorComponent(const AZStd::string& frameId);
+
+        //////////////////////////////////////////////////////////////////////////
```

Comment:
Please remove those `//////////////` from the contribution.
Please only write :
``` FooComponents overrides ...```